Client/GeneralizedImageFetcher.swift

```diff
@@ -0,0 +1,49 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Shared
+import Logger
+
+struct GeneralizedImageFetcher {
+    /// This method is a generalized image fetcher that'll also cache. The completion is dispatched to `main`.
+    func getImageFor(url: URL, completion: @escaping (UIImage?) -> Void) {
+        let generalizedImageTask = URLSession.shared.dataTask(
+            with: URLRequest(
+                url: url,
+                cachePolicy: .reloadRevalidatingCacheData
+            )
+        ) { data, response, error in
+            guard error == nil else {
+                DefaultLogger.shared.log("Error while attempting to fetch image! Error: \(String(describing: error?.localizedDescription))",
+                                         level: .warning,
+                                         category: .core)
```

Comment:
About logging:
- An error in code shouldn't necessarily be an error/warning in log level
- I think this should be debug level. If we have an error from the server in production code (aka, you have made sure already the code works in develop builds) then that image not showing for some users doesn't warrant a warning level. 
- I wouldn't use `.core` here. I'll rename it as this means the name doesn't specify properly it's intent. I would probably create a new category like `images` or something that we'll be able to use in `SiteImageView` since we don't have logs there yet.

Reply:
Gotcha okay - I'll add in `images` as part of this PR and use category.